### PR TITLE
Fixed all non commented kql files and added workflow to validate kql comments.

### PR DESCRIPTION
**Pull request by kpoineal**
# Overview/Summary

Resolved all instance of KQL files not following the prescribed guidance:
- First line // Azure Resource Graph Query
- Second line // <Description>
- Also accounting for variations of "cannot query" due to the ARG query not existing yet.
- [ZT] - Fix issues with linter permissions

Added workflow in kqlfilecheck.yml
- Uses validate-kql-comments.ps1 under .github/scripts to validate the comment structure of each KQL file in the repository
- Ensures KQL files are following the prescribed guidance.
- Runs on PR to main only if a KQL file is modified.

## Related Issues/Work Items

Replace this with a list of related GitHub Issues and/or ADO Work Items (Internal Only)

- To associate a GitHub Issue, use a [key word](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) preceded with the GitHub issue number.
- To associate an ADO Work Item, use the key word `AB#` succeeded with the [ADO Work Item ID](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

## This PR fixes/adds/changes/removes

1. Fixed all KQL files
2. Added workflow for KQL comment validation

### Breaking Changes

1. _Replace me_
2. _Replace me_

## As part of this pull request I have

- [ ] Read the [Contribution Guide](https://azure.github.io/Azure-Proactive-Resiliency-Library-v2/contributing) and ensured this PR is compliant with the guide
- [ ] Checked for duplicate [Pull Requests](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/pulls)
- [ ] Associated it with relevant [GitHub Issues](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/issues) or ADO Work Items (Internal Only)
- [ ] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/tree/main)
- [ ] Ensured PR tests are passing
- [ ] Performed testing and provided evidence (e.g. screenshot of output) for any changes associated to ARG queries
- [ ] Updated relevant and associated documentation (e.g. Contribution Guide, Docs etc.)

### CODE REVIEW

**Comment by kpoineal** [comment]
Oops I never pulled against main before making these changes. We can resolve conflicts by keeping the existing four kql queries for the database accounts.

File: .github/workflows/kqlfilecheck.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    - '**/*.kql'
+  workflow_dispatch: {}
+
+jobs:
+  kql_file_check:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
```
Comment (oZakari) [inline .github/workflows/kqlfilecheck.yml:17]:
```suggestion
        uses: actions/checkout@v4
```

File: azure-resources/Insights/components/kql/dac421ec-2832-4c37-839e-b6dc5a38f2fa.kql [hunk 2]
```diff
@@ -1,3 +1,5 @@
+// Azure Resource Graph query
+// filters Application Insights resources with ‘Classic’ deployment type
```
Comment (oZakari) [inline azure-resources/Insights/components/kql/dac421ec-2832-4c37-839e-b6dc5a38f2fa.kql:2]:
```suggestion
// Filters Application Insights resources with ‘Classic’ deployment type
```

File: azure-resources/Compute/virtualMachineScaleSets/kql/e4ffd7b0-ba24-c84e-9352-ba4819f908c0.kql [hunk 3]
```diff
@@ -1,3 +1,5 @@
+// Azure Resource Graph query
+// Identifies VMs and VMSS with manual patch settings, excluding automatic patch modes, and lists them with a recommendation ID.
```
Comment (oZakari) [inline azure-resources/Compute/virtualMachineScaleSets/kql/e4ffd7b0-ba24-c84e-9352-ba4819f908c0.kql:2]:
```suggestion
// Identifies VMs and VMSS with manual patch settings, excluding automatic patch modes
```

File: azure-resources/Network/virtualNetworkGateways/kql/281a2713-c0e0-3c48-b596-19f590c46671.kql [hunk 4]
```diff
@@ -1,3 +1,5 @@
+// Azure Resource Graph Query
+// identifies non-active-active VPN type virtual network gateways
```
Comment (oZakari) [inline azure-resources/Network/virtualNetworkGateways/kql/281a2713-c0e0-3c48-b596-19f590c46671.kql:2]:
```suggestion
// Identifies non-active-active VPN type virtual network gateways
```

File: azure-resources/Sql/servers/kql/c0085c32-84c0-c247-bfa9-e70977cbf108.kql [hunk 5]
```diff
@@ -1,3 +1,5 @@
+// Azure Resource Graph Query
+// finds non-zone redundant SQL databases and lists them
```
Comment (oZakari) [inline azure-resources/Sql/servers/kql/c0085c32-84c0-c247-bfa9-e70977cbf108.kql:2]:
```suggestion
// Finds non-zone redundant SQL databases and lists them
```

**Review by oZakari — CHANGES_REQUESTED** [review]

**Review by oZakari — APPROVED** [review]

**Review by Dost2010 — APPROVED** [review]

**Pull request merged**